src/jit/lclvars.cpp

```diff
@@ -233,16 +240,55 @@ void Compiler::lvaInitTypeRef()
     InitVarDscInfo varDscInfo;
     varDscInfo.Init(lvaTable, hasRetBuffArg);
 
+    // Maintain a pointer to the start of the arguments.
+    LclVarDsc* varDscTrailingPointer = varDscInfo.varDsc;
     lvaInitArgs(&varDscInfo);
 
     //-------------------------------------------------------------------------
+    // Calculate the argument register usage.
+    //
+    // This will later be used for fastTailCall determination
+    //-------------------------------------------------------------------------
+
+    unsigned   floatingRegCount = 0;
+    unsigned   stackArgCount    = 0;
+    unsigned   stackSize        = 0;
+
+    auto incrementRegCount = [&floatingRegCount, &argRegCount](LclVarDsc* varDsc)
+    {
+        varDsc->IsFloatRegType() ? ++floatingRegCount : ++argRegCount;
+    };
+
+    for (unsigned argNum = 0; argNum < info.compMethodInfo->args.numArgs; argNum++, varDscTrailingPointer++)
```

Comment:
Consider this version of this loop used elsewhere:

`for (lclNum = 0, varDsc = lvaTable; lclNum < lvaCount; lclNum++, varDsc++)`